src/t_zset.c

```diff
@@ -1165,6 +1165,37 @@ unsigned long zsetLength(const robj *zobj) {
     return length;
 }
 
+/* Factory method to return a zset.
+ *
+ * The size hint indicates approximately how many items will be added which is
+ * used to determine the initial representation. */
+robj *zsetTypeCreate(sds value, size_t size_hint) {
+    if (size_hint <= server.zset_max_listpack_entries ||
+        sdslen(value) <= server.zset_max_listpack_value)
```

Comment:
sdslen(value) <= server.zset_max_listpack_value

i think maybe we can remove it? then we can reuse it in zrangeResultBeginStore

Reply:
it is indeed odd to see a `value` passed to this method.
maybe exchange it with a `val_len_hint`, and do the `sdslen` in the caller.
this way you can just pass 0 in zrangeResultBeginStore